### PR TITLE
Testing infrastructure: redis-py 4.x and Valgrind improvements

### PR DESCRIPTION
**Pull request by rafie**

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -58,22 +58,51 @@ commands:
...
 
   setup-prerequisits:
     steps:
       - setup-automation
       - run:
           name: Install prerequisites
-          command: ./sbin/system-setup.py
+          command: |
+            ./sbin/system-setup.py
+            python3 -m pip list
```
Comment (swilly22) [inline .circleci/config.yml:105]: Why do we list installed python packages ?

File: tests/flow/tests.sh [hunk 2]
```diff
@@ -0,0 +1,268 @@
+#!/bin/bash
+
+# [[ $VERBOSE == 1 ]] && set -x
+
+HERE="$(cd "$(dirname "${BASH_SOURCE[0]}")" >/dev/null 2>&1 && pwd)"
+ROOT=$(cd $HERE/../.. && pwd)
+READIES=$ROOT/deps/readies 
+. $READIES/shibumi/defs
+
+VALGRIND_REDIS_VER=6.2
```
Comment (swilly22) [inline tests/flow/tests.sh:10]: wouldn't this become a headache to keep track of ?

File: tests/flow/tests.sh [hunk 3]
```diff
@@ -0,0 +1,268 @@
+#!/bin/bash
```
Comment (swilly22) [inline tests/flow/tests.sh:1]: Why do we need this new file? what's wrong with the current method ?

**Review by swilly22 — CHANGES_REQUESTED** [review]

File: .circleci/config.yml [hunk 4]
```diff
@@ -58,22 +58,51 @@ commands:
...
+  save-tests-logs:
+    steps:
+      - run:
+          name: Cleanup test log dir
+          command: |
+            if [[ -d tests/flow/logs ]]; then
+              cd tests/flow/logs
+              rm -f *.aof *.rdb
+            fi
+            if [[ -d tests/tck/logs ]]; then
```
Comment (chayim) [inline .circleci/config.yml:70]: Why not just remove both. rm -f won't complain:

File: .circleci/config.yml [hunk 5]
```diff
@@ -397,7 +410,7 @@ jobs:
       - run:
           name: Unzip module artifact
           command: |
-            unzip `ls /workspace/$PACKAGE_NAME.Linux-ubuntu18.04-x86_64.*.zip |grep -v rce` -d /workspace/
+            unzip `ls /workspace/$PACKAGE_NAME.Linux-ubuntu18.04-x86_64.*.zip | grep -v rce` -d /workspace/
```
Comment (chayim) [inline .circleci/config.yml:413]: Hardcoded path means other paths in the future are more work, Can this be removed?

File: tests/flow/tests.sh (hunk 2 above)
Comment (chayim) [reply tests/flow/tests.sh:10]: Inject from the Makefile, same as used for docker builds?

Comment (rafie) [reply tests/flow/tests.sh:10]: We don't have one global place in which we state the redis version we use. Here, as we're building a valgrind-comparible redis, we just to be good enough, so we take 6.2. We can also say just `6`, which will yield the same result.
I don't think it creates a maintenance problem (I will go with the `6` thing, though).

File: .circleci/config.yml (hunk 1 above)
Comment (rafie) [reply .circleci/config.yml:105]: Just for information/disgnostics, not essential, can be removed.

File: tests/flow/tests.sh (hunk 3 above)
Comment (rafie) [reply tests/flow/tests.sh:1]: We're moving to a scheme that standardizes driving of flow tests, making it more robust and easy to maintain across modules (it appears that there is a common pattern). It makes it easier to interface RLTest, collect logs, handle valgrind, etc.
We're not using exactly the same script, as the process of convergence is still ongoing and I'm still learning the patterns, but the overall result seems promissing.

File: .circleci/config.yml (hunk 5 above)
Comment (rafie) [reply .circleci/config.yml:413]: This is benchmarking stuff, not really part of this PR. I just added a space there, purely OCD thing.

File: .circleci/config.yml (hunk 4 above)
Comment (rafie) [reply .circleci/config.yml:70]: We want to keep `*.log` and remove `*.aof *.rdb`, which just take space as artifacts and slow down the artifact upload process in CI.
Do you mean like: `rm -f tests/tck/logs/*.aof tests/tck/logs/*.rdb`?

File: tests/flow/tests.sh (hunk 3 above)
Comment (rafie) [reply tests/flow/tests.sh:1]: Note that this change initiated in the RLEC PR, and was also introduced here to support the log games.

File: .circleci/config.yml (hunk 4 above)
Comment (chayim) [reply .circleci/config.yml:70]: Yeah or something like ```rm -f tests/tck/logs/*.{aof|rdb}```

Comment (rafie) [reply .circleci/config.yml:70]: Nice. Will work with a comma: `rm -f tests/tck/logs/*.{aof,rdb}`

File: src/Makefile [hunk 6]
```diff
@@ -15,6 +15,12 @@ ifneq ($(filter cov-upload,$(MAKECMDGOALS)),)
 COV=1
 endif
 
+ifneq ($(filter memcheckk,$(MAKECMDGOALS)),)
```
Comment (swilly22) [inline src/Makefile:18]: `memcheckk` -> `memcheck`

**Review by swilly22 — CHANGES_REQUESTED** [review]

**Review by AviAvni — APPROVED** [review]

**Pull request merged**